cmd/oras/manifest/fetch_config.go

```diff
@@ -0,0 +1,167 @@
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/internal/cache"
+	"oras.land/oras/internal/file"
+)
+
+type fetchConfigOptions struct {
+	option.Common
+	option.Descriptor
+	option.Pretty
+	option.Remote
+
+	cacheRoot  string
+	outputPath string
+	targetRef  string
+}
+
+func fetchConfigCmd() *cobra.Command {
+	var opts fetchConfigOptions
+	cmd := &cobra.Command{
+		Use:   "fetch-config [flag] name<:tag|@digest>",
+		Short: "[Preview] Fetch the config of a manifest from a remote registry",
+		Long: `[Preview] Fetch the config of a manifest from a remote registry
+
+** This command is in preview and under development. **
+
+Example - Fetch the config:
+  oras manifest fetch-config localhost:5000/hello:latest
+
+Example - Fetch and print the prettified config:
+  oras manifest fetch-config --pretty localhost:5000/hello:latest
+
+Example - Fetch the config and save it to a local file:
+  oras manifest fetch-config --output config.json localhost:5000/hello:latest
+
+Example - Fetch the descriptor of the config:
+  oras manifest fetch-config --descriptor localhost:5000/hello:latest
+
+Example - Fetch and print the prettified descriptor of the config:
+  oras manifest fetch-config --descriptor --pretty localhost:5000/hello:latest
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.outputPath == "-" && opts.OutputDescriptor {
+				return errors.New("`--output -` cannot be used with `--descriptor` at the same time")
+			}
+
+			opts.cacheRoot = os.Getenv("ORAS_CACHE")
+			return opts.ReadPassword()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.targetRef = args[0]
+			return fetchConfig(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	option.ApplyFlags(&opts, cmd.Flags())
+	return cmd
+}
+
+func fetchConfig(opts fetchConfigOptions) (fetchErr error) {
+	ctx, _ := opts.SetLoggerLevel()
+
+	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	if err != nil {
+		return err
+	}
+
+	if repo.Reference.Reference == "" {
+		return oerrors.NewErrInvalidReference(repo.Reference)
+	}
+
+	var src oras.ReadOnlyTarget = repo
+	if opts.cacheRoot != "" {
+		ociStore, err := oci.New(opts.cacheRoot)
+		if err != nil {
+			return err
+		}
+		src = cache.New(repo, ociStore)
+	}
+
+	// fetch config descriptor
+	configDesc, err := fetchConfigDesc(ctx, src, opts.targetRef)
+	if err != nil {
+		return err
+	}
+
+	if !opts.OutputDescriptor || opts.outputPath != "" {
+		// fetch config content
+		contentBytes, err := content.FetchAll(ctx, src, configDesc)
+		if err != nil {
+			return err
+		}
+
+		// output config content
+		if opts.outputPath == "" || opts.outputPath == "-" {
+			return opts.Output(os.Stdout, contentBytes)
+		}
+
+		// save config into the local file if the output path is provided
+		if opts.outputPath != "" && opts.outputPath != "-" {
```

Comment:
This `if` statement is redundant since
`!(opts.outputPath == "" || opts.outputPath == "-")` is logically equivalent to `opts.outputPath != "" && opts.outputPath != "-")`.